### PR TITLE
CORTX-30166 - Take system drive as input parameter

### PR DESCRIPTION
**Pull request by shailesh-vaidya**
# Problem Statement
-  We have `/dev/sdb` drive hardcoded in scripts. We need to take it as input for community deployment. This change modifies scripts to take SYSTEM_DRIVE as input. 

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [x] Testing was performed with RPM -  Tested with - https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/RGW/job/setup-cortx-rgw-cluster/2177/console

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Review by AbhijitPatil1992 — APPROVED** [review]
Changes looks good

File: solutions/kubernetes/cortx-deploy.sh [hunk 1]
```diff
@@ -28,38 +28,57 @@ ALL_NODES=$(cat "$HOST_FILE" | awk -F[,] '{print $1}' | cut -d'=' -f2)
...
+    if [ -z "$CORTX_DATA_IMAGE" ]; then echo "CORTX_DATA_IMAGE not provided. Using default : ghcr.io/seagate/cortx-data:2.0.0-latest"; CORTX_DATA_IMAGE=ghcr.io/seagate/cortx-data:2.0.0-latest; fi
+    if [ -z "$DEPLOYMENT_METHOD" ]; then echo "DEPLOYMENT_METHOD not provided. Using default : standard"; DEPLOYMENT_METHOD=standard; fi
+    if [ -z "$SOLUTION_CONFIG_TYPE" ]; then echo "SOLUTION_CONFIG_TYPE not provided. Using default : manual"; SOLUTION_CONFIG_TYPE=manual; fi
     if [ -z "$SNS_CONFIG" ]; then SNS_CONFIG="1+0+0"; fi
     if [ -z "$DIX_CONFIG" ]; then DIX_CONFIG="1+0+0"; fi
     if [ -z "$EXTERNAL_EXPOSURE_SERVICE" ]; then EXTERNAL_EXPOSURE_SERVICE="LoadBalancer"; fi
     if [ -z "$CONTROL_EXTERNAL_NODEPORT" ]; then CONTROL_EXTERNAL_NODEPORT="31169"; fi
     if [ -z "$S3_EXTERNAL_HTTP_NODEPORT" ]; then S3_EXTERNAL_HTTP_NODEPORT="30080"; fi
     if [ -z "$S3_EXTERNAL_HTTPS_NODEPORT" ]; then S3_EXTERNAL_HTTPS_NODEPORT="30443"; fi
+    if [ -z "$SYSTEM_DRIVE" ]; then echo "SYSTEM_DRIVE not provided. Using default: /dev/sdb";SYSTEM_DRIVE="/dev/sdb"; fi
```
Comment (gauravchaudhari02) [inline solutions/kubernetes/cortx-deploy.sh:63]: Are we not adding `SYSTEM_DRIVE` as a Jenkins parameter?

File: solutions/kubernetes/cluster-setup.sh [hunk 2]
```diff
@@ -22,6 +22,8 @@ source functions.sh
 
 HOST_FILE=$PWD/hosts
 SSH_KEY_FILE=/root/.ssh/id_rsa
+validation
```
[the remark this answers is not in the document]
Comment (shailesh-vaidya) [reply solutions/kubernetes/cluster-setup.sh:25]: For `ALL_NODES=$(cat "$HOST_FILE" | awk -F[,] '{print $1}' | cut -d'=' -f2)`  we need to validate that $HOST_FILE is present else it does not give clear error. Hence added validation before fetching node values.

File: solutions/kubernetes/cortx-deploy.sh (hunk 1 above)
Comment (shailesh-vaidya) [reply solutions/kubernetes/cortx-deploy.sh:63]: Not as of now. As Jenkins is only used for Seagate internal systems and we expect SSC VMs with a particular format.

**Review by gauravchaudhari02 — APPROVED** [review]

**Pull request merged**